config/src/main/java/org/axonframework/config/DefaultConfigurer.java

```diff
@@ -1025,5 +1040,10 @@ public EventUpcasterChain upcasterChain() {
         public HandlerDefinition handlerDefinition(Class<?> inspectedType) {
             return handlerDefinition.get().apply(inspectedType);
         }
+
+        @Override
+        public List<ComponentDecorator> getDecorators() {
+            return decorators;
```

Comment:
shouldn't we return a `Collections.unmodifiableList` to make sure the list isn't modified externally?